### PR TITLE
chore: Use github actions for CI and release-please.

### PR DESCRIPTION
**Pull request by kinyoklion**
I will also make a release-please PR and hold merging this until both are ready.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,11 +27,11 @@
   ],
   "scripts": {
     "test": "jest",
-    "test:junit": "jest --ci --reporters=default --reporters=jest-junit",
+    "test:junit": "jest --ci --reporters=default",
     "clean": "rimraf lib/*",
     "rb": "rollup -c --configPlugin typescript",
-    "rbw": "yarn rb --watch",
```
Comment (kinyoklion) [inline package.json:33]: Removing yarn dependency.

File: package.json [hunk 2]
```diff
@@ -60,22 +60,20 @@
...
     "@types/react-dom": "^18.0.0",
     "@types/react-test-renderer": "^18.0.0",
-    "esbuild": "^0.21.3",
-    "jest": "^27.5.1",
-    "jest-environment-jsdom": "^27.4.4",
+"esbuild": "^0.21.3",
+    "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
     "jest-environment-jsdom-global": "^4.0.0",
-    "jest-junit": "^16.0.0",
```
Comment (kinyoklion) [inline package.json:67]: Removing jest-junit as we are not using the result.

File: package.json [hunk 3]
```diff
@@ -60,22 +60,20 @@
...
     "prop-types": "^15.7.2",
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "react-test-renderer": "^18.0.0",
-    "rimraf": "^5.0.7",
+    "rimraf": "^5.0.9",
     "rollup": "^3.26.2",
     "rollup-plugin-dts": "^6.1.0",
     "rollup-plugin-esbuild": "^6.1.1",
-    "rollup-plugin-filesize": "^10.0.0",
```
Comment (kinyoklion) [inline package.json:77]: Removing this plugin. We may want to add a similar plugin in the future, but I am working on minimizing requirements that are not strictly needed at the moment.

File: src/__snapshots__/provider.test.tsx.snap [hunk 4]
```diff
@@ -3,10 +3,10 @@
 exports[`LDProvider render app 1`] = `
 <Provider
   value={
-    Object {
```
Comment (kinyoklion) [inline src/__snapshots__/provider.test.tsx.snap:6]: In Jest 29 the default snapshot formatting changed.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    branches: [main]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-test:
+    strategy:
+      matrix:
+        variations: [
+          # {os: ubuntu-latest, node: latest},
```
Comment (kinyoklion) [inline .github/workflows/ci.yml:18]: Disabled due to this bug: https://github.com/npm/cli/issues/7657

Running on the latest LTS is already a better target, and we can enable this again if that bug is resolved. Or we use a combination of dependencies that doesn't trigger it.

**Review by tanderson-ld — APPROVED** [review]

**Pull request merged**